File: NutritionApp/app/src/main/java/com/bsnutrition/app/core/model/UserProfile.kt
package com.bsnutrition.app.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserProfile(
    @SerialName("user_id") val userId: Long,
    @SerialName("birth_date") val birthDate: String? = null,
    val sex: String? = null,
    val height: Float? = null,
    @SerialName("current_weight") val currentWeight: Float? = null,
    @SerialName("activity_level") val activityLevel: String? = null,
    @SerialName("goal_type") val goalType: String? = null,
    @SerialName("goal_weight") val goalWeight: Float? = null,
    @SerialName("weekly_goal_rate") val weeklyGoalRate: Float? = null,
    val locale: String? = null,
    @SerialName("country_code") val countryCode: String? = null,
    val timezone: String? = null,
    @SerialName("unit_system") val unitSystem: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)

File: nutrition-backend/app/Models/UserProfile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class UserProfile extends Model
{
    use HasFactory;

    protected $table = 'user_profiles';

    protected $fillable = [
        'user_id',
        'birth_date',
        'sex',
        'height',
        'current_weight',
        'activity_level',
        'goal_type',
        'goal_weight',
        'weekly_goal_rate',
        'locale',
        'country_code',
        'timezone',
        'unit_system',
    ];

    /**
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'birth_date' => 'date',
            'height' => 'float',
            'current_weight' => 'float',
            'goal_weight' => 'float',
            'weekly_goal_rate' => 'float',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
